src/global_distrib/mod_global_distrib_server_sup.erl

```diff
@@ -33,9 +33,23 @@ start_link(Server) ->
     SupName = mod_global_distrib_utils:server_to_sup_name(Server),
     supervisor:start_link({local, SupName}, ?MODULE, [Server]).
 
--spec get_connection(Server :: jid:lserver()) -> pid().
+-spec get_connection(Server :: jid:lserver()) -> {ok, pid()} | {error, not_available}.
 get_connection(Server) ->
-    mod_global_distrib_server_mgr:get_connection(Server).
+    case catch mod_global_distrib_server_mgr:get_connection(Server) of
```

Comment:
I would love to see a test for this fix.

Reply:
I'll add a TODO comment here because we would need to have a uniform method of simulating race conditions. Without it, I would be forced to use meck+RPC which is evil and failure-prone.